### PR TITLE
Add info about new contraction data type and permutation feature to README

### PR DESCRIPTION
**Pull request by CongMa13**
Add info about new contraction data type and permutation feature to README

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -98,40 +121,71 @@ After configuration, build with `cmake --build <build_dir> -- -j<nproc>`
...
 Tests the API implementation of scale contraction algorithm with validation.
-o <build_dir>/bin/scale_contraction_f32_test
-o <build_dir>/bin/scale_contraction_f64_test
 
-### Samples
+```bash
+  <build_dir>/bin/scale_contraction_test
+```
+
+## Running Permutation Tests
```
Comment (LisaDelaney) [inline README.md]: This should be sentence case:

```suggestion
## Running permutation tests
```

File: README.md [hunk 2]
```diff
@@ -98,40 +121,71 @@ After configuration, build with `cmake --build <build_dir> -- -j<nproc>`
...
+
+### Permutation tests
```
Comment (LisaDelaney) [inline README.md]: I don't think we need this heading.

File: README.md [hunk 3]
```diff
@@ -98,40 +121,71 @@ After configuration, build with `cmake --build <build_dir> -- -j<nproc>`
...
+
+Tests API implementation of permutation algorithm with validation.
```
Comment (LisaDelaney) [inline README.md]:
```suggestion
Test API implementation of the permutation algorithm with validation.
```

File: README.md [hunk 4]
```diff
@@ -98,40 +121,71 @@ After configuration, build with `cmake --build <build_dir> -- -j<nproc>`
...
+```bash
+  <build_dir>/bin/permutation_test
+```
+
+## Samples
 
 These are stand-alone use-cases of the hipTensor contraction operations.
 
-## F32 Bilinear contraction
+### F32 Bilinear contraction
```
Comment (LisaDelaney) [inline README.md]:
```suggestion
### F32 bilinear contraction
```

File: README.md [hunk 5]
```diff
@@ -98,40 +121,71 @@ After configuration, build with `cmake --build <build_dir> -- -j<nproc>`
...
 
 Demonstrates the API implementation of bilinear contraction operation without validation.
-o <build_dir>/bin/simple_contraction_bilinear_f32
 
-## F32 Scale contraction
+```bash
+  <build_dir>/bin/simple_bilinear_contraction_<typeA>_<typeB>_<typeC>_<typeD>_compute_<computeType>
+```
+
+### F32 Scale contraction
```
Comment (LisaDelaney) [inline README.md]:
```suggestion
### F32 scale contraction
```

File: README.md [hunk 6]
```diff
@@ -98,40 +121,71 @@ After configuration, build with `cmake --build <build_dir> -- -j<nproc>`
...
+
+```bash
+  <build_dir>/bin/simple_permutation
+```
 
 ### Build Samples as external client
 
-Client application links to hipTensor library,
-and therefore hipTensor library needs to be installed before building client applications.
+Client application links to hipTensor library, and therefore hipTensor library
```
Comment (LisaDelaney) [inline README.md]: How about: "The client application links to hipTensor library; therefore, you must install the hipTensor library before building client applications."

File: README.md [hunk 7]
```diff
@@ -98,40 +121,71 @@ After configuration, build with `cmake --build <build_dir> -- -j<nproc>`
 ### Logger tests
 
 Tests API implementation of logger verbosity and functionality.
-o <build_dir>/bin/logger_test
+
+```bash
+  <build_dir>/bin/logger_test
+```
 
 ## Running Contraction Tests
```
Comment (LisaDelaney) [inline README.md]:
```suggestion
## Running contraction tests
```

File: README.md [hunk 8]
```diff
@@ -98,40 +121,71 @@ After configuration, build with `cmake --build <build_dir> -- -j<nproc>`
...
 
 ### Bilinear contraction tests
```
Comment (LisaDelaney) [inline README.md]: Can we make this heading (and Scale contraction tests) into bullet points in order to avoid stacked headings?

**Comment by CongMa13** [comment]
Hi @LisaDelaney , I have made changes you suggested. Could you review it again?

File: README.md [hunk 9]
```diff
@@ -38,15 +38,38 @@ python3 -m sphinx -T -E -b html -d _build/doctrees -D language=en . _build/html
 
 ## Currently supported
 
-Operations - Contraction Tensor
-Data Types - FP32 , FP64
+### Operations - Contraction Tensor
```
Comment (LisaDelaney) [inline README.md:41]: Does "Contractor Tensor" need to be capitalized (i.e. is it a formal name)? All titles should be in sentence case.

Comment (LisaDelaney) [inline README.md]:
```suggestion
### Operation: Contraction tensor
```

File: README.md [hunk 10]
```diff
@@ -38,15 +38,38 @@ python3 -m sphinx -T -E -b html -d _build/doctrees -D language=en . _build/html
...
+| ----------- | ----------- | ----------- | ----------------- | ---------------------------------- |
+| bf16        | bf16        | bf16        | f32               |                                    |
+| __half      | __half      | __half      | f32               |                                    |
+| f32         | f32         | f32         | bf16              |                                    |
+| f32         | f32         | f32         | __half            |                                    |
+| f32         | f32         | f32         | f32               |                                    |
+| f64         | f64         | f64         | f32               | f64 is supported only on gfx90a +  |
+| f64         | f64         | f64         | f64               | f64 is supported only on gfx90a +  |
+
+### Operations - Permutation Tensor
```
Comment (LisaDelaney) [inline README.md]:
```suggestion
### Operation: Permutation tensor
```

File: README.md [hunk 11]
```diff
@@ -69,22 +92,22 @@ git push origin <new_branch>
...
-|Debug build|`CC=hipcc CXX=hipcc cmake -B<build_dir> . -DCMAKE_BUILD_TYPE=Debug` |
-|Build without tests (default on)|`CC=hipcc CXX=hipcc cmake -B<build_dir> . -DHIPTENSOR_BUILD_TESTS=OFF` |
+| Configuration                    | Command                                                                   |
+|----------------------------------|---------------------------------------------------------------------------|
+| Basic                            | `CC=hipcc CXX=hipcc cmake -B<build_dir> .`                                |
+| Targeting gfx908                 | `CC=hipcc CXX=hipcc cmake -B<build_dir> . -DAMDGPU_TARGETS=gfx908:xnack-` |
+| Debug build                      | `CC=hipcc CXX=hipcc cmake -B<build_dir> . -DCMAKE_BUILD_TYPE=Debug`       |
+| Build without tests (default on) | `CC=hipcc CXX=hipcc cmake -B<build_dir> . -DHIPTENSOR_BUILD_TESTS=OFF`    |
 
 After configuration, build with `cmake --build <build_dir> -- -j<nproc>`
```
Comment (LisaDelaney) [inline README.md]:
```suggestion
After configuration, build with `cmake --build <build_dir> -- -j<nproc>`.
```

File: README.md [hunk 12]
```diff
@@ -98,40 +121,69 @@ After configuration, build with `cmake --build <build_dir> -- -j<nproc>`
...
+
+```bash
+  <build_dir>/bin/simple_permutation
+```
 
 ### Build Samples as external client
 
-Client application links to hipTensor library,
-and therefore hipTensor library needs to be installed before building client applications.
+The client application links to hipTensor library; therefore, you must install the
```
Comment (LisaDelaney) [inline README.md]:
```suggestion
The client application links to the hipTensor library; therefore, you must install the
```

File: README.md [hunk 13]
```diff
@@ -98,40 +122,69 @@ After configuration, build with `cmake --build <build_dir> -- -j<nproc>`
...
+
+### Permutation
+
+Demonstrates the API implementation of permutation operation without validation.
+
+```bash
+  <build_dir>/bin/simple_permutation
+```
 
 ### Build Samples as external client
```
Comment (LisaDelaney) [inline README.md]:
```suggestion
### Build samples as external client
```

**Comment by CongMa13** [comment]
Hi @LisaDelaney , the changes you suggested have been merged. Could you review it again?

**Comment by CongMa13** [comment]
Hi @LisaDelaney , @saadrahim ,  could you please approve this PR if it looks good to you?  I can merge it after your approval. Thanks.

File: README.md [hunk 14]
```diff
@@ -9,8 +9,9 @@ through general purpose kernel languages, like HIP C++.
 * AMD CDNA class GPU featuring matrix core support:
 gfx908, gfx90a, gfx940, gfx941, gfx942 as 'gfx9'
 
-> Note: Double precision FP64 datatype support requires
-> gfx90a, gfx940, gfx941 or gfx942
+:::{note}
+Double precision FP64 datatype support requires gfx90a, gfx940, gfx941 or gfx942
+:::
 
 ## Minimum Software Requirements
```
Comment (LisaDelaney) [inline README.md]: Can we make sure all titles have sentence case?
```suggestion
## Minimum software requirements
```

File: README.md [hunk 15]
```diff
@@ -38,15 +39,40 @@ python3 -m sphinx -T -E -b html -d _build/doctrees -D language=en . _build/html
...
-Data Types - FP32 , FP64
+### Operation: Contraction tensor
+
+hiptensor supports a tensor contraction of the form $D = \alpha AB + \beta C$
+
+Supported data-type combinations are:
+
+| typeA       | typeB       | typeC       | typeCompute       | notes                              |
+|-------------|-------------|-------------|-------------------|------------------------------------|
+| bf16        | bf16        | bf16        | f32               |                                    |
```
Comment (LisaDelaney) [inline README.md:48]: Should the letters in this table not be in caps? (e.g., BF16)

File: README.md [hunk 16]
```diff
@@ -38,15 +39,40 @@ python3 -m sphinx -T -E -b html -d _build/doctrees -D language=en . _build/html
...
+| __half      | __half      | __half      | f32               |                                    |
+| f32         | f32         | f32         | bf16              |                                    |
+| f32         | f32         | f32         | __half            |                                    |
+| f32         | f32         | f32         | f32               |                                    |
+| f64         | f64         | f64         | f32               | f64 is supported only on gfx90a +  |
```
Comment (LisaDelaney) [inline README.md]: I would suggest changing "is supported only on" to "is only supported on".

File: README.md (hunk 15 above)
Comment (CongMa13) [reply README.md:48]: I think we can keep the bf16 to be consistent with the RocWMMA's format. 

RocWMMA's readme https://github.com/ROCm/rocWMMA

image

File: README.md (hunk 16 above)
Comment (CongMa13) [reply README.md]: changed to "is only supported on".

**Comment by CongMa13** [comment]
Hi @LisaDelaney ,  the README.md has been updated as you suggested. Could you approve this PR if it's ready to go?

**Review by LisaDelaney — DISMISSED** [review]

**Comment by CongMa13** [comment]
Hi @LisaDelaney , could you approve this PR again. 

Sorry to ask you to do this again. 

Deleted this line from README.md to make it to pass all the checks. 

```
hiptensor supports a tensor contraction of the form $D = \alpha AB + \beta C$
```

Remove it since readthedocs does not accept the Latex expression in this line.

**Comment by LisaDelaney** [comment]
> Hi @LisaDelaney , could you approve this PR again.
> 
> Sorry to ask you to do this again.
> 
> Deleted this line from README.md to make it to pass all the checks.
> 
> ```
> hiptensor supports a tensor contraction of the form $D = \alpha AB + \beta C$
> ```
> 
> Remove it since readthedocs does not accept the Latex expression in this line.

If you want to leave it in, I can find math syntax that works...just let me know.

**Review by LisaDelaney — APPROVED** [review]

**Review by mkarunan — APPROVED** [review]
lgtm

**Pull request merged**